x-pack/plugins/security_solution/common/detection_engine/schemas/request/rule_schemas.ts

```diff
@@ -501,6 +502,10 @@ const responseRequiredFields = {
 
 const responseOptionalFields = {
   execution_summary: RuleExecutionSummary,
+  active_snoozes: t.array(t.string),
+  mute_all: t.boolean,
+  is_snoozed_until: t.union([IsoDateString, t.undefined, t.null]),
+  snooze_schedule: t.array(RuleSnoozeSchedule),
```

Comment:
Let's not expose the full snooze model from the detection_engine API endpoints. It leaks the "implementation" details of this feature from the Alerting Framework to Security:

- As you can see from the diff, you had to add the new fields to a lot of places in the code of `security_solution` plugin.
- It couples the Security code with the Framework code I'd say quite a lot. Every time you need to evolve this model in ResponseOps you will have to remember to tweak the corresponding code on the Security side and align those changes with the Rules and potentially Alerts area. Let's try to avoid this dependency - it seems unnecessary.

My suggestion is to read the full snooze model via an Alerting API endpoint.

- It would be some "bulk get snooze information for multiple rules by their ids", and we'd call it on every page refresh of the Rules table. In addition, our existing API endpoints in Security could be returning just basic information about rule snoozing (e.g. `is_snoozed_now`, `has_snooze_schedule`; I'd probably even deprecate `mute_all` if possible).
- The basic information that would be returned with rules could be easily constructed on the BE side based on the full snooze model, probably using some helper methods of the `RulesClient`. This info could be helpful to make the initial rendering of the Rules Table, Rule Details page, etc -- this could reduce the number of spinners and show _some_ minimally useful info right away.
- The full snoozing information would be fetched via a dedicated Alerting endpoint and used to re-render the Snooze components to show full info and make them interactive.

Reply:
Yes that make sense to me too and I even like it better